File: Medium/Min Product Subset/min-product-subset.cpp
class Solution {
  public:
  
    int n;
  
    int solve(int ind , int prod , bool taken ,vector<int>& arr){
        if (ind == n){
            if (!taken) return INT_MAX;
            return prod;
        }
        
        int take = solve(ind+1 , prod * arr[ind] , true , arr);
        
        int leave = solve(ind+1 , prod , taken , arr);
        
        return min(take , leave);
        
    }
  
    int minProd(vector<int>& arr) {
        // code here
        n = arr.size();
        return solve(0 , 1 , false , arr);
        
    }
};

File: Medium/Min Product Subset/min-product-subset.py
class Solution:
    def minProd(self, arr):
        
        n = len(arr)
        
        # code here
        def solve(ind , prod , selected):
            if ind == n:
                if selected:
                    return prod
                return float('inf')
                
            take = solve(ind+1 , prod*arr[ind] , True)
            
            leave = solve(ind+1 , prod , selected)
            
            return min(take , leave)
            
            
            
        return solve(0 , 1 , False)
        